### ISSUE TITLE
[Bug] New hyperparameters logged through HParam don't get displayed in Tensorboard

### ISSUE THREAD

**Issue by AlexPasqua** [DLR-RM/stable-baselines3#1099]
### 🐛 Bug

Using the [HParam logger for Tensorboard](https://stable-baselines3.readthedocs.io/en/master/guide/tensorboard.html#logging-hyperparameters), if, in a new run, I add a new hyperparameter to be logged, I don't see it in tensorboard.
I see it only if the first run in the log directory was already logging that hyperparameter.

### To Reproduce

Log some runs on Tensorboard and then log a new one which keeps track of a new hyperparameter using the [HParam logger for Tensorboard](https://stable-baselines3.readthedocs.io/en/master/guide/tensorboard.html#logging-hyperparameters),

**Example:**
Say we have [the logging callback from SB3's docs](https://stable-baselines3.readthedocs.io/en/master/guide/tensorboard.html#logging-hyperparameters), which I copy here for simplicity.
```python
from stable_baselines3 import A2C
from stable_baselines3.common.callbacks import BaseCallback
from stable_baselines3.common.logger import HParam


class HParamCallback(BaseCallback):
    def __init__(self):
        """
        Saves the hyperparameters and metrics at the start of the training, and logs them to TensorBoard.
        """
        super().__init__()

    def _on_training_start(self) -> None:
        hparam_dict = {
            "algorithm": self.model.__class__.__name__,
            "learning rate": self.model.learning_rate,
            "gamma": self.model.gamma,
        }
        # define the metrics that will appear in the `HPARAMS` Tensorboard tab by referencing their tag
        # Tensorbaord will find & display metrics from the `SCALARS` tab
        metric_dict = {
            "rollout/ep_len_mean": 0,
            "train/value_loss": 0,
        }
        self.logger.record(
            "hparams",
            HParam(hparam_dict, metric_dict),
            exclude=("stdout", "log", "json", "csv"),
        )

    def _on_step(self) -> bool:
        return True


model = A2C("MlpPolicy", "CartPole-v1", tensorboard_log="runs/", verbose=1)
model.learn(total_timesteps=int(5e4), callback=HParamCallback())
```

Now run a training of a model, and on Tensorboard, in the `HPARAMS` tab, you'll see a row corresponding to that run containing the value for "algorithm", "learning rate", "gamma" etc.

Now add a new hyperparameter to the `hparam_dict` of the method `_on_training_start` and perform another training execution. On Tensorboard, you'll se a new row corresponding to the new execution, but the new hyperparameter won't be displayed.

### Expected behavior

I'd expect the new hyperparameter to be logged and have its own column on Tensorboard, with the cells belonging to rows which correspons to runs where this hyperparameter wasn't logged that stay empty, a bit like in the following image, where the "GCN layers dimensions" column is empty in the second row:

![image](https://user-images.githubusercontent.com/33461106/194156224-b3cdceaf-ccee-44ce-914c-b4a77554187a.png)

### Notes

- The previous image was generated executing **first** a training logging all the hyperparameters and **then** another run where the log of "GCN layers dimensions" was disabled.
- I'm aware that on the left of the Tensorboard HPARAMS tab there is a menu where you can select the quantities to display, but the new hyperparameter won't show up there as well.

### System Info

Output of `sb3.get_system_info()`:

```
OS: Windows-10-10.0.19044-SP0 10.0.19044
Python: 3.8.10
Stable-Baselines3: 1.6.1
PyTorch: 1.12.1+cpu
GPU Enabled: False
Numpy: 1.23.3
Gym: 0.21.0
```

 * Tensorboard version: 2.10.0
   * `tensorboard-plugin-wit` version: 1.8.1
   * `tensorboard-data-server` version: 0.6.1

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)
- [x] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)

**Comment by araffin** [comment]
@timothe-chaumont do you have an idea? (original pr was https://github.com/DLR-RM/stable-baselines3/pull/984)

**Comment by araffin** [comment]
closing as the issue comes from PyTorch.
